snapshot/snapshotter.go

```diff
@@ -282,6 +285,38 @@ func checkSnapshotterBasic(t *testing.T, snapshotter Snapshotter, work string) {
 		t.Fatal(err)
 	}
 
+	fooMode := os.FileMode(0777)
+	aMode, bMode, cMode := os.ModeDir|0755, os.ModeDir|0755, os.ModeDir|0755
+	info1 := &testutil.FileInfo{
+		BaseName: "",
+		// If you set Mode to os.ModeDir | 0777 for "", overlay fails (btrfs and naive passes)
```

Comment:
The root is not usually included in these types of comparisons.

As for overlay, my understanding is overlay will overwrite the permission of the mount location with the root of the overlay. Tried changing the permission of the root to match the expectations but still getting a failure.